### PR TITLE
[DT-538][risk=no] Implement has_ehr_data for cohort builder criteria

### PR DESCRIPTION
**Pull request by chenchals**
---
- Implemented `has_ehr_data` selection under `Demographics`
- Added SQL to get counts for  `has_ehr_data`
- Indexing build nedds to be run in test environment after this PR is merged
- I have manually tested the change to `cd_criteria_menu` in local docker environment  with
  - `"enableHasEhrData": true` 
https://github.com/all-of-us/workbench/assets/4452480/49cb466c-8bbc-4f50-8fd2-220fb16456dd
  - `"enableHasEhrData": false`
  

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/pages/data/cohort/cohort-search.tsx [hunk 1]
```diff
@@ -352,6 +354,23 @@ export const CohortSearch = fp.flow(
       this.setState({ toastVisible: true });
     };
 
+    selectHasEhrData() {
+      const param = {
+        id: null,
+        parentId: null,
+        parameterId: '',
+        type: CriteriaType.HAS_EHR_DATA.toString(),
+        name: 'Deceased',
```
Comment (dolbeew) [inline ui/src/app/pages/data/cohort/cohort-search.tsx:363]: Copy/paste typo

File: api/src/main/resources/workbench-api.yaml [hunk 2]
```diff
@@ -4863,7 +4863,7 @@ definitions:
       enableRStudioGKEApp:
         type: boolean
         default: false
-        description: Whether users are able to use the RStudio GKE app.
+        description: Whether users are able to use the RStudio GKE app./criteria
```
Comment (freemabd) [inline api/src/main/resources/workbench-api.yaml:4866]: Is this a typo?

File: ui/src/app/pages/data/cohort/cohort-search.tsx (hunk 1 above)
Comment (freemabd) [inline ui/src/app/pages/data/cohort/cohort-search.tsx:363]: This should not be 'Deceased'

**Review by dolbeew — COMMENTED** [review]
Also need to add a case for `HAS_EHR_DATA` to the [typeToTitle function](https://github.com/all-of-us/workbench/blob/chenchals/DT-538-hasEhrData/ui/src/app/pages/data/cohort/utils.ts#L142) for the search group item display.

**Review by freemabd — CHANGES_REQUESTED** [review]
Left to comments that need changes.

File: api/src/main/resources/workbench-api.yaml (hunk 2 above)
Comment (chenchals) [reply api/src/main/resources/workbench-api.yaml:4866]: doh... was thinking I was searching for /criteria :-). Fixed it. Thanks

File: ui/src/app/pages/data/cohort/cohort-search.tsx (hunk 1 above)
Comment (chenchals) [reply ui/src/app/pages/data/cohort/cohort-search.tsx:363]: copy-paste -> fixed

**Review by dolbeew — APPROVED** [review]

**Pull request merged**